### PR TITLE
[Docs] Updates `definition-of-done.md`

### PR DESCRIPTION
**Pull request by mnigh**
## 👋 Introduction

This PR updates the Definition of Done document to include items that closer mirror what is actually being considered currently. Suggestions are very welcome!

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Review and ensure copy in `definition-of-done.md` file makes sense.

### CODE REVIEW

File: documentation/definition-of-done.md [hunk 1]
```diff
@@ -1,20 +1,16 @@
...
-- Step by step testing instructions
-- At least one peer review acceptance
-- Chromatic testing by design
-- Definition of done checklist from issue template (eventually)
-- If the route to a live page has been changed, a Permanent Redirect (301) has also been added to the nginx conf file
-- Build pipeline successful, including:
-  - JavaScript linting
+- Step-by-step testing instructions
+- At least one peer review approval
+- If the route to a live page has been changed, a Permanent Redirect (301) has also been added to the `nginx` conf files
```
Comment (petertgiles) [inline documentation/definition-of-done.md:5]: Hmm, this came up recently.  It's easy and useful for static URLs but for URLs with a dynamic component, not so much.

File: documentation/definition-of-done.md [hunk 2]
```diff
@@ -1,20 +1,16 @@
...
+- Build pipeline successful, including passing tests for:
```
Comment (petertgiles) [inline documentation/definition-of-done.md:6]: Is there value in listing our CI steps here?  They run automatically - you don't need a checklist.

File: documentation/definition-of-done.md [hunk 3]
```diff
@@ -1,20 +1,16 @@
-# Definition of Done:
+# Definition of Done
```
Comment (petertgiles) [inline documentation/definition-of-done.md:1]: I always forget what this document represents.  Worth adding a sentence or two to introduce it?

**Review by petertgiles — COMMENTED** [review]
Very nice!  I don't know how far you want to go with this edit but I just listed some thoughts.

File: documentation/definition-of-done.md (hunk 2 above)
Comment (mnigh) [reply documentation/definition-of-done.md:6]: Agree! I thought the same thing but didn't want to go too wild here, but I will!

File: documentation/definition-of-done.md (hunk 3 above)
Comment (mnigh) [reply documentation/definition-of-done.md:1]: That is a great question. **Definition of Done** is an agile term and I am not sure we are actually following what agile suggests to include 😨.

Comment (petertgiles) [reply documentation/definition-of-done.md:1]: A quick google search takes me to https://www.leadingagile.com/2017/02/definition-of-done/ .  It's interesting that it notes that the DoD is different at different stages.  The description in the "Feature" section sounds quite similar (to me) to what we're aiming for.

Comment (petertgiles) [reply documentation/definition-of-done.md:1]: Also, it reminds me that "completing the acceptance criteria" should probably be on the list!  :laughing:

File: documentation/definition-of-done.md (hunk 1 above)
Comment (petertgiles) [reply documentation/definition-of-done.md:5]: Also, this should probably be part of the acceptance criteria, not implied in the DoD.

**Review by esizer — APPROVED** [review]
This is great, very concise 👍

**Pull request merged**